### PR TITLE
Changes for 0.12.0

### PR DESCRIPTION
**Pull request by doolse**
Still waiting for joncfoo/purescript-parsing#update-for-0.12.0 to be merged in though

Didn't attempt to change the anything to use `CodePoint` instead of `Char` or do anything `unsafeDecoderURIComponent`

### CODE REVIEW

**Comment by vyorkin** [comment]
The build is failing here https://github.com/shinnn/install-purescript/blob/master/index.js#L146, I see that we're using node v6.0 on travis CI, if I understand correctly *object spread* is supported only since nodejs v8.6.0 https://node.green/#ES2018-features-object-rest-spread-properties-object-spread-properties. I'm not sure about this, but you could try to update a version of nodejs in the travis CI config file and run the build again.

P.S.: I need this package too, migrating to 0.12.0 :)

**Comment by vyorkin** [comment]
btw it makes sense to remove the import of `Monoid` in `Path.purs` because it has been moved into the Prelude https://github.com/purescript/purescript-prelude/releases/tag/v4.0.0

**Comment by nsaunders** [comment]
For what it's worth, I had the *object spread* issue that @vyorkin mentioned in [paf31/purescript-quickserve#8](https://github.com/paf31/purescript-quickserve/pull/8), and upgrading to Node 9 solved it. It definitely looks to me like the same thing is happening here.

Aside from that, I just want to add that I too am eagerly awaiting upgrading a dependent library, so I would love to know if there is anything I can do to help out here... :-)

File: test/Main.purs [hunk 1]
```diff
@@ -2,28 +2,27 @@ module Test.Main where
...
 import Test.URI.UserInfo as UserInfo
-import Test.Util (TestEffects)
 
-main ∷ Eff (TestEffects (avar ∷ AVAR, process ∷ PROCESS)) Unit
+
+main ∷ Effect Unit
 main = run [consoleReporter] do
   Scheme.spec
-  UserInfo.spec
+  UserInfo.spec
```
Comment (safareli) [inline test/Main.purs]: space

**Comment by i-am-the-slime** [comment]
Any chance this might get merged despite the odd extraneous space?

**Comment by garyb** [comment]
Thanks! Sorry about the delay, I pushed a few more minor changes too.

**Pull request merged**